libraries/chain/apply_context.cpp

```diff
@@ -184,10 +184,13 @@ void apply_context::exec_one()
       r.auth_sequence[auth.actor] = next_auth_sequence( auth.actor );
    }
 
-   trx_context.executed_action_receipt_digests.emplace_back( r.digest() );
-
    finalize_trace( trace, start );
 
+   if (trx_context.executed_action_receipts.digests_l)
+      trx_context.executed_action_receipts.digests_l->emplace_back( trace.digest_legacy() );
+   if (trx_context.executed_action_receipts.digests_s)
+      trx_context.executed_action_receipts.digests_s->emplace_back( trace.digest_savanna() );
```

Comment:
What about having `trx_context.executed_action_receipts.append(trace)` and have `action_digests_t ` call the appropriate `digest_*` method and assign to appropriate digests.

Reply:
trace is a `transaction_trace`, and we compute digests for `action_trace` receipts.

Reply:
Beautiful! I named the function `trx_context.executed_action_receipts.compute_and_append_digests_from(trace);`